src/rpcserver.cc

```diff
@@ -590,28 +580,25 @@ void RPCServer::handlePacket(const uint8_t *buf, size_t buflen, const SocketAddr
             return;
         }
 
-        /* Maybe error these codes.
-            *
         // 1. the message is not a request
         // 2. transaction ID matched
         // 3. request destination did not match response source!!
         // this happening by chance is exceedingly unlikely
         // indicates either port-mangling NAT, a multhomed host listening on any-local address or some kind of attack
         // ignore response
         log->warn("Transaction id matched, socket address did not, ignoring message, request: {} -> response: {}, version: {}",
-                call.getRequest().getRemote(), msg.getOrigin(), msg.getReadableVersion());
+                call->getRequest()->getRemoteAddress().toString(), msg->getOrigin().toString(), msg->getReadableVersion());
 
-        if(msg->getType() == Message::Type::RESPONSE && dht->getType() == DHT.Type.IPV6) {
+        if(msg->getType() == Message::Type::RESPONSE && dht6) {
             // this is more likely due to incorrect binding implementation in ipv6. notify peers about that
             // don't bother with ipv4, there are too many complications
-            Message err = new ErrorMessage(msg.getMethod(), msg.getTxid(), ErrorCode.ProtocolError.value(),
-                    "A request was sent to " + call.getRequest().getRemote() +
-                    " and a response with matching transaction id was received from " + msg.getOrigin() +
+            auto err = std::make_shared<ErrorMessage>(msg->getMethod(), msg->getTxid(), ErrorCode::ProtocolError,
```

Comment:
why need std::make_shared to create a ErrorMessage from heap memory, can we just definte an ErrorMessage variable from stack ?
ErrorMessage error(msg->getMethod(), .....)